packages/workflow/src/interfaces.ts

```diff
@@ -438,10 +438,10 @@ export type Handler<
   Ret,
   Args extends any[],
   T extends SignalDefinition<Args> | QueryDefinition<Ret, Args>
-> = T extends SignalDefinition<infer A>
-  ? (...args: A) => void | Promise<void>
-  : T extends QueryDefinition<infer R, infer A>
-  ? (...args: A) => R
+> = T extends SignalDefinition<Args>
+  ? (...args: Args) => void | Promise<void>
+  : T extends QueryDefinition<Ret, Args>
+  ? (...args: Args) => Ret
```

Comment:
`infer` was unnecessary here, since the type parameters in the true branch of the conditional are not narrowed w.r.t. to the input type parameters. And in fact, `infer` was harmful, because instead of using information about the non-emptiness of `Args` at the point where the `Handler` type was used, it was essentially pattern-matching against the `SignalDefinition` / `QueryDefinition` types and thus incorrectly permitting empty `Args`.